processor/src/main/java/org/mapstruct/ap/spi/AccessorNamingStrategy.java

```diff
@@ -15,6 +17,16 @@
  */
 public interface AccessorNamingStrategy {
 
+    /**
+     * Initializes the accessor naming strategy with the utility methods for operating on elements and types.
+     *
+     * @param elementUtils utility methods for operating on elements
+     * @param typeUtils utility methods for operating on types
+     */
+    default void init(Elements elementUtils, Types typeUtils) {
```

Comment:
Should we wrap both `Elements` and `Types` in a bean so that we can extend it without changing the signature with other utils when needed in the future? I'm not sure myself here...

Reply:
I was actually thinking the same and was not sure what the best approach is. I'll wrap in in some bean so we are more flexible in the future. By the way we might need to do something similar for the `BuilderProvider` (just so we are consistent)

Reply:
@sjaakd just updated this. Can you have another look? Not 100% sure about the name of the new interface though

Reply:
you mean `  public void init(MapStructProcessingEnvironment processingEnvironment) `.. LGTM.. What are you doubting about?

Reply:
The name of `MapStructProcessingEnvironment ` of the interface. Just pushed new changes it should be OK to add a new default method to an interface now (`japicmp-maven-plugin` was marking that as binary incompatible)